var/spack/repos/builtin/packages/mofem-users-modules/package.py

```diff
@@ -0,0 +1,97 @@
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+import os
+
+
+class MofemUsersModules(CMakePackage):
+    """mofem users modules"""
+
+    homepage = "http://mofem.eng.gla.ac.uk"
+    url = "https://bitbucket.org/likask/mofem-joseph/downloads/users_modules_dummy"
+    version('1.0', '5a8b22c9cdcad7bbad92b1590d55edb1', expand=False)
+
+    maintainers = ['likask']
+
+    variant('copy_user_modules', default=True,
+        description='Copy user modules directory instead linking')
+    variant('with_metaio', default=False,
+        description='Install MetaIO with MoFEM users modules')
+
+    variant('mofem_fracture_module', default=False,
+        description="Install fracture mechanics module")
+    variant('mofem_minimal_surface_equation', default=False,
+        description="Install minimal surface equation module")
+
+    extends('mofem-cephas')
+
+    depends_on("mofem-fracture-module", type=('build', 'link', 'run'),
+        when='+mofem_fracture_module')
+    depends_on("mofem-minimal-surface-equation", type=('build', 'link', 'run'),
+        when='+mofem_minimal_surface_equation')
+
+    @property
+    def root_cmakelists_dir(self):
+        """The relative path to the directory containing CMakeLists.txt
+
+        This path is relative to the root of the extracted tarball,
+        not to the ``build_directory``. Defaults to the current directory.
+
+        :return: directory containing CMakeLists.txt
+        """
+        spec = self.spec
+        return os.path.join(spec['mofem-cephas'].prefix.users_modules)
+
+    @property
+    def build_directory(self):
+        """Returns the directory to use when building the package
+
+        :return: directory where to build the package
+        """
+        return os.path.join(self.prefix, 'build')
+
+    def cmake_args(self):
+        spec = self.spec
+
+        options = []
+
+        # obligatory options
+        options.extend([
+            '-DWITH_METAIO=%s' % ('YES' if '+with_metaio' in spec else 'NO'),
+            '-DSTAND_ALLONE_USERS_MODULES=%s' %
+            ('YES' if '+copy_user_modules' in spec else 'NO')])
+
+        ext_um_modules_opt = '-DEXTERNAL_MODULE_SOURCE_DIRS='
+        if '+mofem_fracture_module' in spec:
+            ext_um_modules_opt += '%s;' % \
+                spec['mofem-fracture-module'].prefix.ext_users_modules
+
+        if '+mofem_minimal_surface_equation' in spec:
+            ext_um_modules_opt += '%s;' % \
+                spec['mofem-minimal-surface-equation'].prefix.ext_users_modules
+
+        options.append(ext_um_modules_opt)
+
+        return options
+
+    phases = ['cmake', 'build']
```

Comment:
since you actually build in `os.path.join(self.prefix, 'build')` i guess that's your _installation_, kind of? 

If so, i would rather build where Spack suggest by default and then install binaries by hand into `self.prefix`. For example, see https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/superlu-dist/package.py#L109-L120

Reply:
I've checked this. Unfortunately, this won't work. All paths will point to wrong directories.

It is a key that the user could recompile in view when he changes something in the code or adds a new module.

Reply:
This bit compiling directly to install prefix in not perfect. However all works pretty well. I tested that on computer cluster centos7, ubuntu and mac.